### PR TITLE
feat: autoevent adds buffer for sending event to coredata

### PR DESCRIPTION
**Pull request by weichou1229**
Signed-off-by: weichou <weichou1229@gmail.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Committing+Code+Guidelines#CommittingCodeGuidelines-CommitMessages
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #657 

## What is the new behavior?
Add buffer and initial with the AsyncBufferSize for handling the auto event.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: internal/autoevent/executor.go [hunk 1]
```diff
@@ -77,7 +77,12 @@ func (e *Executor) Run(ctx context.Context, wg *sync.WaitGroup, dic *di.Containe
 				if event.Origin == 0 {
 					event.Origin = common.GetUniqueOrigin()
 				}
-				go common.SendEvent(event, lc, container.CoredataEventClientFrom(dic.Get))
+
+				go func() {
```
Comment (lenny-intel) [inline internal/autoevent/executor.go:81]: Please add comment describing `why` this is needed.

**Review by lenny-intel — CHANGES_REQUESTED** [review]
LGTM

File: internal/autoevent/executor.go (hunk 1 above)
Comment (weichou1229) [reply internal/autoevent/executor.go:86]: Added comment.

Comment (cloudxxx8) [reply internal/autoevent/executor.go:85]: @lenny-intel do you think the comment is acceptable?

Comment (cloudxxx8) [reply internal/autoevent/executor.go:85]: @lenny-intel could you please review this PR again?

**Comment by cloudxxx8** [comment]
@lenny-intel this is blocked by your change request.  could you please take a look?

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**